### DISCUSSION TITLE
Temporarily hide or disable Starship prompt

### DISCUSSION THREAD

**Discussion by mloskot** [starship/starship#5804] [Q&A] [answered]
Hi,
Is there any way to temporarily disable the Starship prompt and restore the original shell's native prompt?

Perhaps there is an environment variable like `STARSHIP_DISABLE=true` that could be exported to bring back the original prompt?

I think this would be useful when one needs to take a screenshot and don't want to 'leak' Azure subscription details. Kubernetes context details,  or just wants to make it cleaner getting rid of the fancy prompt data not relevant for the purpose for which screenshot is being taken.

**Comment by matoruru** [comment, marked as answer]
@mloskot You can use `starship toggle <module>` to disable the specific module on the fly, like `starship toggle kubernetes`, `starship toggle azure`, ...

But if you need to disable whole starship, you can have the condition like this

```
[ "$STARSHIP_DISABLE" = "true" ] || <Your starship init command>
```

in your `.bashrc` or `config.fish` whatever files.

And open the new shell process with no starship by `STARSHIP_DISABLE=true bash` 😀

**Reply by mloskot** [reply]
The `starship toggle` is what does the on-demand trick for me. Thank you!

**Reply by aryklein** [reply]
is there any way to toggle automatically at completing the `kubectl` command?
[powerlevel10k](https://github.com/romkatv/powerlevel10k) works in this way

**Comment by mloskot** [comment]
There is a brute-force way to disable Starship:

```bash
export STARSHIP_CONFIG=
```

**Comment by abclements** [comment]
If you want a plain bash shell you could always just: `alias command="bash -norc -c [command]"` and that will give you the system default bash prompt while using whatever shell command and then when it closes you have the starship prompt back.
